app/payments.py

```diff
@@ -126,63 +126,58 @@ def get_payments(id):
         result['error'] = result['error'].format(id)
         rc = HTTP_404_NOT_FOUND
 
+
     return make_response(jsonify(result), rc)
 
 ######################################################################
 # UPDATE AN EXISTING PAYMENT
 ######################################################################
 @app.route('/payments/<int:id>', methods=['PUT'])
 def update_payments(id):
-    index = [i for i, payment in enumerate(payments) if payment['id'] == id]
-    if len(index) > 0:
+    try:
+        if not request.data:
+            raise DataValidationError('Invalid payment: body of request contained bad or no data')
         if not request.is_json:
-            return make_response(CONTENT_ERR_MSG, HTTP_400_BAD_REQUEST)
-        payload = request.get_json()
-        if is_valid(payload):
-            payments[index[0]] = {'id' : id, 'nickname' : payload['nickname'], 'type' : payload['type'],
-                                  'default' : payments[index[0]]['default'],
-                                  'charge-history' : payments[index[0]]['charge-history'],
-                                  'detail' : payload['detail']}
-            message = payments[index[0]]
-            rc = HTTP_200_OK
-        else:
-            message = { 'error' : 'Payments data was not valid' }
-            rc = HTTP_400_BAD_REQUEST
-    else:
-        message = { 'error' : 'Payments %s was not found' % id }
+            raise DataValidationError('Invalid payment: body of request contained bad or no data')
+        try:
```

Comment:
i think we could satisfy both of these `if` blocks by cutting them out and just starting with:

`data = request.get_json(silent=True)`

if request doesn't exist OR if not json, this will "fail silently" meaning it won't raise an exception but simply return None to payload. Then we could either pass `None` to update_payments inside a try/except block and let it handle the error and raise it back if need be.

or let `get_json` throw its error here where we can catch it and do something with it. though i think it would throw a 404 and get picked up by the error handler if nothing else.

specific implementation up to you, but either way, we can cut down on some of this extra code